app/routes/titiler/algorithms.py

```diff
@@ -26,25 +62,47 @@ def __call__(self, img: ImageData) -> ImageData:
         """Encode Integrated alerts to RGBA."""
         mask = img.array.mask[0].astype(int)
         data = img.data[0]
+        alert_date = data % 10000  # in days since 2014-12-31
+        data_alert_confidence = data // 10000
+
+        r = np.zeros_like(data_alert_confidence, dtype=int)
+        g = np.zeros_like(data_alert_confidence, dtype=int)
+        b = np.zeros_like(data_alert_confidence, dtype=int)
+
+        for _, properties in ALERT_CONFIDENCE_MAP.items():
+            confidence = properties["confidence"]
+            colors = properties["colors"]
+
+            r[data_alert_confidence >= confidence] = colors["red"]
+            g[data_alert_confidence >= confidence] = colors["green"]
+            b[data_alert_confidence >= confidence] = colors["blue"]
+
+        start_mask = alert_date >= (
+            np.datetime64(self.start_date).astype(int)
+            - np.datetime64(START_DATE).astype(int)
+        )
+        end_mask = alert_date <= (
+            np.datetime64(self.end_date).astype(int)
+            - np.datetime64(START_DATE).astype(int)
+        )
+
+        confidence_mask = (
+            data_alert_confidence
+            >= ALERT_CONFIDENCE_MAP[self.alert_confidence]["confidence"]
+        )
 
-        # Will add this in next iteration
-        # if self.start_date:
-        #     start_mask = data % 10000 >= (
-        #         np.datetime64(self.start_date).astype(int) - self.START_DATE
-        #     )
-        #     mask = mask * start_mask
-        # if self.end_date:
-        #     end_mask = data % 10000 <= (
-        #         np.datetime64(self.end_date).astype(int) - self.START_DATE
-        #     )
-        #     mask = mask * end_mask
-
-        r = np.where(data > 0, 228, 0)
-        g = np.where(data > 0, 102, 0)
-        b = np.where(data > 0, 153, 0)
-
-        intensity = np.where(mask == 0, img.data[1], 0)
-        intensity = np.minimum(255, intensity * 50)
+        intensity = np.where(
```

Comment:
You only apply the masks to intensity and not the rgb bands. Is this because you're just setting the alpha to zero for any masked out pixel anyway?

Reply:
Yup